docs/source/tutorial_metrics.md

```diff
@@ -0,0 +1,372 @@
+```
+
+The global metric works well, but have some drawbacks: if we were to start
+training on a multitask datasets, we would not be able to distinguish the
+`global_hello` of the two datasets, and we could only compute the micro-average
+of the combination of the two. Below is an excerpt from a training log with
+the above agents:
+
+```
+09:14:52 | time:112s total_exs:90180 epochs:0.41
+                clip  ctpb  ctps  exps  exs  global_hello  gnorm  gpu_mem  loss  lr  ltpb  ltps   ppl  token_acc  total_train_updates   tpb   tps   ups
+   all             1  9831 66874 841.9 8416        .01081  2.018    .3474 5.078   1  1746 11878 163.9      .2370                  729 11577 78752 6.803
+   convai2                             3434        .01081                 5.288                 197.9      .2120
+   dailydialog                         4982        .01081                 4.868                   130      .2620
+```
+
+Notice how `global_hello` is the same in both, because the model is unable to
+distinguish between the two settings. In the next section we'll show how to fix
+this with local metrics.
+
+__On placement__: In the example above, we recorded the global metric inside
+the `observe` function. However, global metrics can be recorded from anywhere.
+
+### Local metrics
+
+Having observed the limitation of global metrics being unable to distinguish
+settings in multitasking, we would like to improve upon this. Let's add a local
+metric, which is recorded _per example_. By recording this metric per example,
+we can unambiguously identify which metrics came from which dataset, and report
+averages correctly.
+
+Local metrics have a limitation: they can only be computed inside the scope of
+`batch_act`. This includes common places like `compute_loss` or `generate`,
+where we often want to instrument specific behavior.
+
+Let's look at an example. We'll add a metric inside the `batchify` function,
+which is called from within `batch_act`, and is used to convert from a list of
+[Messages](messages) objects to a
+[Batch](torch_agent.html#parlai.core.torch_agent.Batch) object. It is where we do things like
+padding, etc. We'll do something slightly different than our previous runs.
+In this case, we'll count the number of _tokens_ which are the word "hello".
+
+```python
+from parlai.core.metrics import AverageMetric
+from parlai.core.loader import register_agent
+from parlai.agents.transformer.transformer import TransformerGeneratorAgent
+
+
+@register_agent('LocalHelloCounter')
+class LocalHelloCounterAgent(TransformerGeneratorAgent):
+    def batchify(self, observations):
+        batch = super().batchify(observations)
+        if hasattr(batch, 'text_vec'):
+            num_hello = ["hello" in o['text'] for o in observations]
+            self.record_local_metric(
+                'local_hello',
+                # AverageMetric.many(seq1) is shorthand for
```

Comment:
nit: seq1 -> seq same naming as the next line.

Reply:
ty